sharding/proposer/service.go

```diff
@@ -23,6 +23,7 @@ type Proposer struct {
 	client       *mainchain.SMCClient
 	shardp2p     sharding.ShardP2P
```

Comment:
See 2 comments below about discontinuing interfaces for things with a single implementation. Just use *p2p.Server and *txpool.ShardTXPool - #177.